components/hash/node/index.mjs

```diff
@@ -1,9 +1,10 @@
 import { createHash } from "node:crypto";
 
-export const hashFile = ({ url, content }) => {
+export const hashStringArray = (strings) => {
   const hash = createHash("sha256");
-  hash.update(url, "utf8");
-  hash.update("\0", "utf8");
-  hash.update(content, "utf8");
+  for (const string of strings) {
+    hash.update(string, "utf8");
+    hash.update("\0", "utf8");
+  }
```

Comment:
Note this generic construction is not cryptographically safe, since generic strings can include null characters, which makes it possible for two different inputs to hash the same. It's unlikely to be relevant here, but if you want to improve it, you should hash *(length, string)* pairs instead.

Reply:
I added another commit for that since we already use null byte as a separator before this PR:  f292848c413668e6cea8a746e77fb4536e9c03b3

Reply:
Before this PR it was ok, since `url` presumably couldn't contain null characters. Making the algorithm more generic is what introduced the problem. Anyway, thanks for the fix!